GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -2196,96 +2202,108 @@ private void CommandsToolStripMenuItem_DropDownOpening(object sender, EventArgs
+            if (e.CloseReason == ToolStripDropDownCloseReason.ItemClicked)
+            {
+                e.Cancel = true;
+            }
         }
 
-        private void SetDefaultPullActionIfNeeded(AppSettings.PullAction actionKind)
+        private void setAsDefaultToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            if (AppSettings.SetNextPullActionAsDefault)
+            SetDefaultAction(sender, setPullAsDefaultToolStripMenuItem, AppSettings.PullAction.None);
+            SetDefaultAction(sender, setMergeAsDefaultToolStripMenuItem, AppSettings.PullAction.Merge);
+            SetDefaultAction(sender, setRebaseAsDefaultToolStripMenuItem, AppSettings.PullAction.Rebase);
+            SetDefaultAction(sender, setFetchAsDefaultToolStripMenuItem, AppSettings.PullAction.Fetch);
+            SetDefaultAction(sender, setFetchAllAsDefaultToolStripMenuItem, AppSettings.PullAction.FetchAll);
+            SetDefaultAction(sender, setFetchAndPruneAsDefaultToolStripMenuItem, AppSettings.PullAction.FetchPruneAll);
+
+            RefreshDefaultPullAction();
+
+            void SetDefaultAction(object s, ToolStripMenuItem menuItem, AppSettings.PullAction pullAction)
             {
-                AppSettings.DefaultPullAction = actionKind;
-                RefreshPullIcon();
+                if (ReferenceEquals(s, menuItem))
+                {
+                    AppSettings.DefaultPullAction = pullAction;
+                    menuItem.Checked = true;
+                }
+                else
+                {
+                    menuItem.Checked = false;
+                }
             }
         }
 
         private void mergeToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            SetDefaultPullActionIfNeeded(AppSettings.PullAction.Merge);
-
             PullToolStripMenuItemClick(sender, e);
         }
 
         private void rebaseToolStripMenuItem1_Click(object sender, EventArgs e)
         {
-            SetDefaultPullActionIfNeeded(AppSettings.PullAction.Rebase);
-
             PullToolStripMenuItemClick(sender, e);
         }
 
         private void fetchToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            SetDefaultPullActionIfNeeded(AppSettings.PullAction.Fetch);
-
             PullToolStripMenuItemClick(sender, e);
         }
 
         private void pullToolStripMenuItem1_Click(object sender, EventArgs e)
         {
-            SetDefaultPullActionIfNeeded(AppSettings.PullAction.None);
-
             PullToolStripMenuItemClick(sender, e);
         }
 
-        private void RefreshPullIcon()
+        private void RefreshDefaultPullAction()
         {
             switch (AppSettings.DefaultPullAction)
             {
                 case AppSettings.PullAction.Fetch:
                     toolStripButtonPull.Image = Images.PullFetch;
                     toolStripButtonPull.ToolTipText = _pullFetch.Text;
+                    setFetchAsDefaultToolStripMenuItem.Checked = true;
```

Comment:
We set, but never reset this flag. Is it fine?

Reply:
The only other place where .Checked for these submenus is changed is when a default item is changed and there all options are reset according to newly selected default so at least at this moment there is no need to do that here.

Reply:
Got it, thanks :)